refactor: M3-7034 - Event Messages Part 2 (#10550)
linode/manager: docs/development-guide/15-api-events.md

  - the primary action: (ex: `<strong>created</strong>`) 
  - its correlated negation for negative actions (ex: `could <strong>not</strong> be <strong>created</strong>.`)
- The `message` should be also handled via the `<EventMessage message={e.message} />` in order to handle potential formatting from the API string (ticks to indicate code blocks).

> - The message composition can be enhanced by using custom components. If for instance we need to fetch extra data based on an event entity, we can simple write a new component to include in the message:

Reviewer: ```suggestion
- The message composition can be enhanced by using custom components. For instance, if we need to fetch extra data based on an event entity, we can simply write a new component to include in the message:
```

```Typescript
export const linode: PartialEventMap<'linode'> = {
  linode_migrate_datacenter: {
    started: (e) => <LinodeMigrateDataCenterMessage event={e} />,
  },
};

const LinodeMigrateDataCenterMessage = ({ event }: { event: Event }) => {
  const { data: linode } = useLinodeQuery(event.entity?.id ?? -1);
  const { data: regions } = useRegionsQuery();
  const region = regions?.find((r) => r.id === linode?.region);

  return (
    <>
      Linode <EventLink event={event} to="entity" /> is being{' '}
      <strong>migrated</strong>
      {region && (
        <>
          {' '}
          to <strong>{region.label}</strong>
        </>
      )}
      .
    </>
  );
};
```

## In Progress Events

> Some events messages are meant to be displayed alongside a progress bar to show the user the percentage of the action's completion. When an action is in progress, the polling interval switches to every two seconds to provide the user with real-time feedback.

Reviewer: ```suggestion
Some event messages are meant to be displayed alongside a progress bar to show the user the percentage of the action's completion. When an action is in progress, the polling interval switches to every two seconds to provide real-time feedback.
```

> Despite receiving a `percent_complete` value from the API, not all actions are suitable for displaying visual progress, often time because they're too short, or we ony receive 0% and 100% from the endpoint. To allow only certain events to feature the progress bar, their action keys must be added to the `ACTIONS_TO_INCLUDE_AS_PROGRESS_EVENTS` constant.

Reviewer: ```suggestion
Despite receiving a `percent_complete` value from the API, not all actions are suitable for displaying visual progress, often because they're too short, or we only receive 0% and 100% from the endpoint. To allow only certain events to feature the progress bar, their action keys must be added to the `ACTIONS_TO_INCLUDE_AS_PROGRESS_EVENTS` constant.
```
